main.go

```diff
@@ -137,7 +137,10 @@ func monitorNetworkState(testNetwork *network.Network) (awResultsWriters []*csv.
 				config.NodesCount,
 				relevantValidators,
 			)
-
+			if Max(Max(opinions[multiverse.Blue], opinions[multiverse.Red]), opinions[multiverse.Green]) > config.SimulationStopThreshold*config.NodesCount {
+				log.Info("Shutting down simulation (consensue reached) ... [DONE]")
+				os.Exit(0)
```

Comment:
Use channel to send shutdown signal to the main thread and replace sleep in line 50 with waiting for the shutdown signal.

Reply:
and it would be nice to specify max simulation time with time.After that will also terminate the program